src/contrib/cluster/Akka.DistributedData/Replicator.cs

```diff
@@ -419,6 +423,14 @@ public Replicator(ReplicatorSettings settings)
             else Become(NormalReceive);
         }
 
+        private IImmutableList<Address> NodesForReadWrite()
+        {
+            if (_settings.PreferOldest)
+                return _membersByAge.Select(i => i.Address).ToImmutableList();
+            else
+                return _nodes.ToImmutableList();
+        }
```

Comment:
How frequently is this method called, out of curiosity?

Reply:
if you're worried about the `ToImmutableList` being called to often, another option for `PreferOldest==false` scenario would be to switch to `IReadOnlyList` instead of `IImmutableList`. Then we could skip it

Reply:
We can optimize it in a subsequent PR if it's an issue